Source/RapyutaSimulationPlugins/Private/Robots/Turtlebot3/TurtlebotBurger.cpp

```diff
@@ -6,70 +6,81 @@
-        CasterBack->SetupAttachment(Base);
+    Base = CreateDefaultSubobject<UStaticMeshComponent>(TEXT("Base"));
+    LidarSensor = CreateDefaultSubobject<UStaticMeshComponent>(TEXT("LidarSensor"));
+    LidarComponent = CreateDefaultSubobject<URR2DLidarComponent>(TEXT("LidarComp"));
+    WheelLeft = CreateDefaultSubobject<UStaticMeshComponent>(TEXT("WheelLeft"));
+    WheelRight = CreateDefaultSubobject<UStaticMeshComponent>(TEXT("WheelRight"));
+    CasterBack = CreateDefaultSubobject<UStaticMeshComponent>(TEXT("CasterBack"));
 
-        // Constraints
-        Base_LidarSensor = CreateDefaultSubobject<UPhysicsConstraintComponent>(TEXT("Base_LidarSensor"));
-        Base_LidarSensor->SetupAttachment(LidarSensor);
+    check(RootComponent);
+    Base->SetupAttachment(RootComponent);
+    LidarSensor->SetupAttachment(Base);
+    LidarComponent->SetupAttachment(LidarSensor);
+    WheelLeft->SetupAttachment(Base);
+    WheelRight->SetupAttachment(Base);
+    CasterBack->SetupAttachment(Base);
+    bBodyComponentsCreated = true;
 
-        Base_WheelLeft = CreateDefaultSubobject<UPhysicsConstraintComponent>(TEXT("Base_WheelLeft"));
-        Base_WheelLeft->SetupAttachment(WheelLeft);
+    // Constraints
+    Base_LidarSensor = CreateDefaultSubobject<UPhysicsConstraintComponent>(TEXT("Base_LidarSensor"));
+    Base_LidarSensor->SetupAttachment(LidarSensor);
 
-        Base_WheelRight = CreateDefaultSubobject<UPhysicsConstraintComponent>(TEXT("Base_WheelRight"));
-        Base_WheelRight->SetupAttachment(WheelRight);
+    Base_WheelLeft = CreateDefaultSubobject<UPhysicsConstraintComponent>(TEXT("Base_WheelLeft"));
+    Base_WheelLeft->SetupAttachment(WheelLeft);
 
-        Base_CasterBack = CreateDefaultSubobject<UPhysicsConstraintComponent>(TEXT("Base_CasterBack"));
-        Base_CasterBack->SetupAttachment(CasterBack);
+    Base_WheelRight = CreateDefaultSubobject<UPhysicsConstraintComponent>(TEXT("Base_WheelRight"));
+    Base_WheelRight->SetupAttachment(WheelRight);
 
-        IsInitialized = true;
+    Base_CasterBack = CreateDefaultSubobject<UPhysicsConstraintComponent>(TEXT("Base_CasterBack"));
+    Base_CasterBack->SetupAttachment(CasterBack);
 
-        SetupConstraintsAndPhysics();
-        SetupWheels();
-    }
+    SetupConstraintsAndPhysics();
+}
+
+void ATurtlebotBurger::OnConstruction(const FTransform& InTransform)
+{
+    Super::OnConstruction(InTransform);
+    SetupWheelDrives();
 }
 
-void ATurtlebotBurger::SetupWheels()
+void ATurtlebotBurger::SetupWheelDrives()
 {
-    if (IsInitialized)
+    if (bBodyComponentsCreated)
     {
-        UDifferentialDriveComponent* DifferentialDriveComponent = Cast<UDifferentialDriveComponent>(RobotVehicleMoveComponent);
-        DifferentialDriveComponent->SetWheels(Base_WheelLeft, Base_WheelRight);
-        DifferentialDriveComponent->SetPerimeter();
+        UDifferentialDriveComponent* diffDriveComponent = CastChecked<UDifferentialDriveComponent>(RobotVehicleMoveComponent);

```

Comment:
Ah yeap, I intentionally make it Camel lowercase since it is a local variable, versus a member one as Pascal. This has been added to Oss for consistency, but so far mostly my own preference only.

Reply:
pleae add to document.